server/platform/services/sharedchannel/service_api.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sharedchannel
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+)
+
+// ShareChannel marks a local channel as shared. If the channel is already shared this method has
+// no effect and returns without error.
+// TeamId, type, displayname, purpose, and header are fetched from the channel if not provided.
+func (scs *Service) ShareChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
+	channel, err := scs.server.GetStore().Channel().Get(sc.ChannelId, true)
+	if err != nil {
+		return nil, fmt.Errorf("cannot fetch channel while sharing channel %s: %w", sc.ChannelId, err)
+	}
+
+	// check if channel is already shared
+	scExisting, err := scs.server.GetStore().SharedChannel().Get(sc.ChannelId)
+	if err == nil {
+		// already shared, nothing to do
+		return scExisting, nil
+	}
+	if !isNotFoundError(err) {
+		return nil, fmt.Errorf("cannot check if channel %s is shared: %w", sc.ChannelId, err)
+	}
+
+	if sc.TeamId == "" {
+		sc.TeamId = channel.TeamId
+	}
+	if sc.Type == "" {
+		sc.Type = channel.Type
+	}
+	if sc.ShareName == "" {
+		sc.ShareName = channel.Name
+	}
+	if sc.ShareDisplayName == "" {
+		sc.ShareDisplayName = channel.DisplayName
+	}
+	if sc.SharePurpose == "" {
+		sc.SharePurpose = channel.Purpose
+	}
+	if sc.ShareHeader == "" {
+		sc.ShareHeader = channel.Header
+	}
+	if sc.CreatorId == "" {
+		sc.CreatorId = channel.CreatorId
+	}
+
+	// stores the SharedChannel and sets the share flag on the channel.
+	scNew, err := scs.server.GetStore().SharedChannel().Save(sc)
+	if err != nil {
+		return nil, err
+	}
+
+	scs.notifyClientsForSharedChannelConverted(channel)
+	return scNew, nil
+}
+
+// UpdateSharedChannel updates the shared channel details such as displayname, purpose, or header
+func (scs *Service) UpdateSharedChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
+	channel, err := scs.server.GetStore().Channel().Get(sc.ChannelId, true)
+	if err != nil {
+		return nil, err
+	}
+
+	scUpdated, err := scs.server.GetStore().SharedChannel().Update(sc)
+	if err != nil {
+		return nil, err
+	}
+
+	scs.notifyClientsForSharedChannelUpdate(channel)
+	return scUpdated, nil
+}
+
+// UnshareChannel unshared the channel by deleting the SharedChannels record and unsets the Channel `shared` flag.
+// Returns true if a shared channel existed and was deleted.
+func (scs *Service) UnshareChannel(channelID string) (bool, error) {
+	channel, err := scs.server.GetStore().Channel().Get(channelID, true)
+	if err != nil {
+		return false, err
+	}
+
+	// deletes the ShareChannel, unsets the share flag on the channel, deletes all remotes for the channel
+	deleted, err := scs.server.GetStore().SharedChannel().Delete(channelID)
```

Comment:
"All remotes"? To confirm, if we happen to share a channel with "Shared channels" (the feature) and then stop sharing, will that also stop the plugin from getting sync messages?

Reply:
I've clarified the comment.  If the plugin calls `UnshareChannel` then it will no longer receive sync messages for *that* channel because the `SharedChannelRemotes` record will be deleted that maps remotes to channels and tracks the cursor.  This will not prevent the plugin from receiving sync messages for the channels it remains subscribed to.

 ```suggestion
	// deletes the ShareChannel, unsets the share flag on the channel, deletes all records in SharedChannelRemotes for the channel.
	deleted, err := scs.server.GetStore().SharedChannel().Delete(channelID)
```